### PR TITLE
Audits `XrpTrustSet` Protobuf Wrapper

### PR DESCRIPTION
**Pull request by mvadari**
## High Level Overview of Change

This PR adds additional logic validations and descriptive error throwing to the class `XrpTrustSet`.

### Context of Change

Another component of our attempt to make protobuf wrapper classes as safe and informative as possible. An SDK safety net.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

## Before / After

`XrpTrustSet` performs logic/error checking, based on the business logic in the docs.

## Test Plan

New tests added to confirm that errors are correctly thrown. Existing tests were edited to properly account for the new errors. CI passes.

## Future Tasks
This is part of a series, with all subtasks listed here: https://app.asana.com/0/1188384817180008/1193021408531788/f

### CODE REVIEW

File: src/XRP/model/xrp-trust-set.ts [hunk 1]
```diff
@@ -16,14 +17,21 @@ export default class XrpTrustSet {
...
-      return undefined
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'TrustSet protobuf missing required field `LimitAmount`.',
+      )
     }
     const limitAmount = XrpCurrencyAmount.from(limitAmountCurrencyAmount)
-    if (!limitAmount) {
-      return undefined
+    if (!limitAmount.issuedCurrency) {
```
Comment (amiecorso) [inline src/XRP/model/xrp-trust-set.ts:30]: nice

**Review by amiecorso — APPROVED** [review]
look great

File: src/XRP/model/xrp-trust-set.ts [hunk 2]
```diff
@@ -16,14 +17,21 @@ export default class XrpTrustSet {
    * @return an XrpTrustSet with its fields set via the analogous protobuf fields.
    * @see https://github.com/ripple/rippled/blob/3d86b49dae8173344b39deb75e53170a9b6c5284/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L312
    */
-  public static from(trustSet: TrustSet): XrpTrustSet | undefined {
+  public static from(trustSet: TrustSet): XrpTrustSet {
     const limitAmountCurrencyAmount = trustSet.getLimitAmount()?.getValue()
+    // limitAmountCurrencyAmount is required
```
Comment (tedkalaw) [inline src/XRP/model/xrp-trust-set.ts:22]: imo error message is clear enough

**Review by tedkalaw — APPROVED** [review]
gogogo

**Pull request merged**